source/CorpusImpl.cpp

```diff
@@ -291,12 +283,11 @@ class CorpusImpl::
     void visit(NamespaceInfo& I) override
     {
         postProcess(I);
-        canonicalize(I.Children.Namespaces);
-        canonicalize(I.Children.Records);
-        canonicalize(I.Children.Functions);
-        canonicalize(I.Children.Typedefs);
-        canonicalize(I.Children.Enums);
-        canonicalize(I.Children.Vars);
+        canonicalize(I.Members);
+        // KRYSTIAN FIXME: should we canonicalize specializations?
+        // we shouldn't canonicalize anything if we intend to
+        // preserve declaration order.
```

Comment:
I would say we canonicalize in the Generator not the Corpus but that's a bigger change